lib/routes/account.js

```diff
@@ -843,6 +848,7 @@ module.exports = function (
         var operation = payload.id ? 'updateDevice' : 'createDevice'
         db[operation](sessionToken.uid, sessionToken.tokenId, payload).then(
           function (device) {
+            push.notifyNewDevice(sessionToken.uid, device.name, device.id.toString('hex'))
             reply(butil.unbuffer(device))
```

Comment:
You should probably do the reply first to avoid slowing down the requests. Then do the push stuff.